addons/weapononback/functions/fnc_onDrag.sqf

```diff
@@ -0,0 +1,90 @@
+#include "script_component.hpp"
+/*
+ * Author: BaerMitUmlaut
+ * Handles dragging weapons onto the secondary slot.
+ *
+ * Arguments:
+ * 0: Drag origin <CONTROL>
+ * 1: Information about the dragged item <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+ */
+params ["_ctrl", "_info"];
+
+if !(secondaryWeapon ACE_player in ["", QGVAR(weapon)]) exitWith {};
+
+// Check if dragged item is a primary weapon
+private _ctrlIDC = ctrlIDC _ctrl;
+private _isPrimary = if (_ctrlIDC == IDC_PRIMARY_SLOT) then {
+    (primaryWeapon ACE_player != "") && {_info == 0}
+} else {
+    // Dragged from container
+    _info#0 params ["_displayName"];
+
+    private _container = [_ctrlIDC] call FUNC(getIDCContainer);
+    private _index = (weaponCargo _container) findIf {
+        private _cfg = configFile >> "CfgWeapons" >> _x;
+        (getText (_cfg >> "displayName") == _displayName)
+        && {getNumber (_cfg >> "type") == TYPE_WEAPON_PRIMARY}
+    };
+
+    _index != -1
+};
+
+if (!_isPrimary) exitWith {
+    TRACE_1("Dragged item is not a primary",_info#0);
+};
+
+// Set up drag and drop catcher
+private _display = ctrlParent _ctrl;
+private _slotSecondary = _display displayCtrl IDC_SECONDARY_SLOT;
+
+// RscPicture does not trigger LBDrop
+private _dropCatcher = _display ctrlCreate ["ctrlActivePictureKeepAspect", IDC_DROP_CATCHER];
+if (secondaryWeapon ACE_player != QGVAR(weapon)) then {
+    _dropCatcher ctrlSetText QPATHTOF(data\weapononback_ca.paa);
+} else {
+    _dropCatcher ctrlSetText "#(rgb,8,8,3)color(1,1,1,0)";
```

Comment:
`#(rgb,8,8,3)color(1,1,1,x)` maybe should be replaced with macro here and below

Reply:
I considered that, but maybe we should add that in CBA instead?

Reply:
He meant macro because you use the same one twice.

Reply:
I know and I meant that such a macro would fit into CBA 😜

Reply:
I wouldn't wait for it in CBA, first do it in ACE, then (some day) add to cba, then wait for cba release, them move to cba